### PR TITLE
OIDC Blog Post

### PR DESCRIPTION
**Pull request by meagancojocar**

### CODE REVIEW

File: themes/default/content/blog/oidc-blog/index.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+# At least one tag is required. Lowercase, hyphen-delimited is recommended.
+tags:
+    - features
+    - pulumi-releases
+
+# See the blogging docs at https://github.com/pulumi/pulumi-hugo/blob/master/BLOGGING.md.
+# for additional details, and please remove these comments before submitting for review.
+---
+
+After [launching Pulumi Deployments](/blog/pulumi-deployments) a few months ago and receiving a ton of community feedback (thank you!) we are launching today an integration with OpenID Connect (OIDC) to enable temporary credentials and granular access controls. We would like to thank GitHub user [eriklz](https://github.com/eriklz) for creating the [original request](https://github.com/pulumi/service-requests/issues/144) for this functionality.
```
Comment (EvanBoyle) [inline themes/default/content/blog/oidc-blog/index.md:37]: nit: "today we are launching"

Comment (EvanBoyle) [reply themes/default/content/blog/oidc-blog/index.md:37]: Might also be worth mentioning that the OIDC integration is specifically for AWS, Azure, and GCP.

File: themes/default/content/blog/oidc-blog/index.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+## Set up
+
+### Calling the Deployments REST API
+
+You can call the [Deployments REST API](/docs/reference/deployments-rest-api) and use OIDC instead of cloud secrets. Here is an example of how that might look:
+
+![Deployment REST API payload before and after](deployment-api-payload.png)
+
+### Enabling OIDC for your Stack
```
Comment (EvanBoyle) [inline themes/default/content/blog/oidc-blog/index.md:58]: Title might clarify that this is only required for Push to Deploy. There is no configuration required for API usage other than specifying the appropriate params in the POST payload.

**Review by EvanBoyle — APPROVED** [review]
Overall this looks great! Thank you for putting this together. Added a few small comments but it is good to go IMO.

File: themes/default/content/blog/oidc-blog/index.md (hunk 1 above)
Comment (pgavlin) [reply themes/default/content/blog/oidc-blog/index.md:37]: > Might also be worth mentioning that the OIDC integration is specifically for AWS, Azure, and GCP.

This is a bit of a subtle point. We have OIDC integration that automatically fetches creds for those platforms, but we also support OIDC more generally by making our OIDC token available in the `PULUMI_OIDC_TOKEN` environment variable and the `/mnt/pulumi/pulumi.oidc` file. Users can consume the tokens manually in order to integrate with other platforms.

Comment (meagancojocar) [inline themes/default/content/blog/oidc-blog/index.md:37]:
```suggestion
After [launching Pulumi Deployments](/blog/pulumi-deployments) a few months ago and receiving a ton of community feedback (thank you!) today we are launching an integration with OpenID Connect (OIDC) to enable temporary credentials and granular access controls. We would like to thank GitHub user [eriklz](https://github.com/eriklz) for creating the [original request](https://github.com/pulumi/service-requests/issues/144) for this functionality.
```

File: themes/default/content/blog/oidc-blog/index.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+<!--more-->
+
+[OpenID Connect (OIDC)](https://openid.net/connect) is an identity layer built on top of the OAuth 2.0 framework. It allows third-party applications to verify the identity of the end-user and to obtain basic user profile information. OIDC uses JSON web tokens (JWTs), which you can obtain using flows conforming to the OAuth 2.0 specifications.
+
+## Overview
+
+The main benefits of using the OIDC integration for Pulumi Deployments are as follows:
+
+1. **Temporary credentials**: You can move away from long term credentials within your organization to leverage temporary credentials that are generated every time a Deployment runs. The Pulumi Service issues a new OIDC token specific to that deployment and then it automatically expires, meaning you do not duplicate your cloud credentials and instead can rotate credentials for each deployment.
```
Comment (pgavlin) [inline themes/default/content/blog/oidc-blog/index.md:47]:
```suggestion
1. **Temporary credentials**: You can move away from long term credentials within your organization to leverage temporary credentials that are generated every time a deployment runs. The Pulumi Service issues a new OIDC token that is limited to that deployment. This token is used to obtain cloud credentials that automatically expire, meaning you do not need to store sensitive credentials that are reused with each deployment.
```

File: themes/default/content/blog/oidc-blog/index.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+2. **Granular with your access control**: The Pulumi Service will generate the OIDC token, a signed JWT that contains information about the deployment, to be exchanged for cloud credentials. In AWS for example, you can configure our OIDC integration to only use temporary credentials if that token is associated with a particular Pulumi stack or Pulumi organization. This enables you to follow the principle of least privilege access and scope access down to only what is needed.
```
Comment (pgavlin) [inline themes/default/content/blog/oidc-blog/index.md:48]:
```suggestion
2. **Granular access control**: The OIDC token generated by the Pulumi Service contains detailed information about the deployment with which it is associated. In AWS for example, you can configure our OIDC integration to only use temporary credentials if that token is associated with a particular Pulumi stack or Pulumi organization. This enables you to follow the principle of least privilege access and scope access down to only what is needed.
```

File: themes/default/content/blog/oidc-blog/index.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+## Set up
```
Comment (pgavlin) [inline themes/default/content/blog/oidc-blog/index.md:50]:
```suggestion
## Setup
```

**Review by pgavlin — COMMENTED** [review]
A few comments, but looks great overall. Thanks for putting this together!

File: themes/default/content/blog/oidc-blog/index.md (hunk 1 above)
Comment (pgavlin) [reply themes/default/content/blog/oidc-blog/index.md]: Let me give it a re-read and see if and where it this should be captured.

File: themes/default/content/blog/oidc-blog/index.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+
+## Overview
+
+The main benefits of using the OIDC integration for Pulumi Deployments are as follows:
+
+1. **Temporary credentials**: You can move away from long term credentials within your organization to leverage temporary credentials that are generated every time a deployment runs. The Pulumi Service issues a new OIDC token that is limited to that deployment. This token is used to obtain cloud credentials that automatically expire, meaning you do not need to store sensitive credentials that are reused with each deployment.
+2. **Granular access control**: The OIDC token generated by the Pulumi Service contains detailed information about the deployment with which it is associated. In AWS for example, you can configure our OIDC integration to only use temporary credentials if that token is associated with a particular Pulumi stack or Pulumi organization. This enables you to follow the principle of least privilege access and scope access down to only what is needed.
+
```
Comment (pgavlin) [inline themes/default/content/blog/oidc-blog/index.md:49]:
```suggestion
For AWS, Azure, and GCP, Pulumi Deployments can automatically exchange its OIDC token for temporary credentials and make those credentials available to providers. For advanced scenarios or other cloud providers, the OIDC token can be manually exchanged for credentials by the user.
```

Comment (pgavlin) [reply themes/default/content/blog/oidc-blog/index.md:49]: How about this to describe the automatic integration for AWS, Azure, and GCP?

**Pull request merged**